chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -696,18 +711,23 @@ func (n *NeutrinoNotifier) RegisterConfirmationsNtfn(txid *chainhash.Hash,
 
 	ntfn := &confirmationsNotification{
 		ConfNtfn: chainntnfs.ConfNtfn{
+			ID:               atomic.AddUint64(&n.confClientCounter, 1),
 			TxID:             txid,
 			NumConfirmations: numConfs,
 			Event:            chainntnfs.NewConfirmationEvent(numConfs),
 		},
 		heightHint: heightHint,
 	}
 
+	if err := n.txConfNotifier.Register(&ntfn.ConfNtfn); err != nil {
```

Comment:
Can we just move this call to where we are handling the `confirmationsNotification`, before we start the historical dispatch?

Reply:
We can, but I figured it made more sense having it here like we do with `NotifySpent` with `RegisterSpendNtfn`.

Reply:
For readability I think it would make sense to move it there, as they always need to be called in that order, right after one another, right?

Reply:
`Register` must now be called first. Since it's possible that it fails, we should return the error to the caller rather than just logging it, as otherwise the caller will be expecting to receive the confirmation details for a failed registration. At the moment, it's unlikely for it to return an error other than when shutting down, but it could prevent an obscure bug later down the road.